avoinspector/src/development/java/app/avo/inspector/VisualInspector.java

```diff
@@ -0,0 +1,110 @@
+package app.avo.inspector;
+
+import android.app.Activity;
+import android.app.Application;
+
+import androidx.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import app.avo.androidanalyticsdebugger.DebuggerManager;
+import app.avo.androidanalyticsdebugger.DebuggerMode;
+import app.avo.androidanalyticsdebugger.EventProperty;
+import app.avo.androidanalyticsdebugger.PropertyError;
+
+public class VisualInspector {
+    @Nullable
+    DebuggerManager debugger;
+
+    public VisualInspector(AvoInspectorEnv env, Application application, Activity rootActivityForVisualInspector) {
+        if (env != AvoInspectorEnv.Prod) {
+            debugger = new DebuggerManager(application);
+            if (rootActivityForVisualInspector != null) {
```

Comment:
If this can be null - then there should be an annotation for the function parameter.
I also suggest adding this: https://www.jetbrains.com/help/idea/parametersarenonnullbydefault-annotation.html so that every parameter is NonNull unless suggested otherwise. The difference between Kotlin and Java is that in Java you can never be sure if a parameter can be null or not without annotations.

Reply:
Makes sense, but I think just throwing in the default non null is a bit dangerous, I would prefer to have nullable as default (like it is in Java) and mark NonNull fields.